Formula/gfan.rb

```diff
@@ -55,31 +49,332 @@ def install
+--- a/src/enumeration.cpp
++++ b/src/enumeration.cpp
+@@ -14,7 +14,7 @@ EnumerationFilePrinter::EnumerationFilePrinter():
+ 
+ EnumerationFilePrinter::~EnumerationFilePrinter()
+ {
+-  assert(file==0);
++  // assert(file==0);
+ }
+ 
+ 
+diff --git a/src/enumeration.h b/src/enumeration.h
+index b0bd187..d91c9ff 100644
+--- a/src/enumeration.h
++++ b/src/enumeration.h
+@@ -36,7 +36,7 @@ class EnumerationFilePrinter: public EnumerationPrinter
+   FILE *file;
+  public:
+   EnumerationFilePrinter();
+-  ~EnumerationFilePrinter();
++  virtual ~EnumerationFilePrinter();
+ 
+   void open(std::string filename);
+   void open(FILE *file);
+diff --git a/src/field.h b/src/field.h
+index 866be22..7df1600 100644
+--- a/src/field.h
++++ b/src/field.h
+@@ -208,6 +208,7 @@ class FieldElementImplementation
+   {
+ 	  fprintf(stderr,"*this is not in Z/pZ.\n");
+ 	  assert(0);
++	  exit(0);
+   }
+   virtual bool isInteger()const
+   {
+@@ -223,6 +224,7 @@ class FieldElementImplementation
+   Field& operator=(const Field& a)
+     {
+       assert(0);
++      exit(1);
+     }//assignment
+ };
+ 
+@@ -271,7 +273,8 @@ class FieldImplementation
+   virtual FieldElement random()
+   {
+ 	  assert(0);
+-  }
++	  exit(1);
++       }
+     virtual int getCharacteristic()const=0;
+     virtual const char *name()=0;
+   virtual std::string toString()const=0;
+diff --git a/src/field_rationalfunctions2.cpp b/src/field_rationalfunctions2.cpp
+index 4e80c5a..ccc6ae9 100644
+--- a/src/field_rationalfunctions2.cpp
++++ b/src/field_rationalfunctions2.cpp
+@@ -524,6 +524,7 @@ class FieldElementRationalFunction2 : public FieldElementImplementation
+ 
+     return sign+string("{").append(s,startIndex,slashIndex-startIndex)+string("\\over ").append(s,slashIndex+1,s.length()-slashIndex-1)+string("}");
+ */
++	  exit(1);
+ 	  }
+ 
+   std::string toString(bool writeIfOne=true, bool alwaysWriteSign=false, bool latexMode=false /*, bool mathMode=true*/) const
+diff --git a/src/field_rationals.cpp b/src/field_rationals.cpp
+index 30a92f6..06c95c9 100644
+--- a/src/field_rationals.cpp
++++ b/src/field_rationals.cpp
+@@ -398,7 +398,7 @@ const char *FieldRationalsImplementation::name()
+ /*FieldRationals::FieldRationals():
+   Field(new FieldRationalsImplementation())
+ {
+-  /*  fprintf(Stderr,"Adding field rationals\n");
++  fprintf(Stderr,"Adding field rationals\n");
+   next=list;
+   list=this;
+   */
+diff --git a/src/gfanlib_circuittableint.h b/src/gfanlib_circuittableint.h
+index 2b5ced4..609d2d2 100644
+--- a/src/gfanlib_circuittableint.h
++++ b/src/gfanlib_circuittableint.h
+@@ -25,6 +25,7 @@ namespace gfan{
+   template<typename> struct MyMakeUnsigned;
+   template <> struct MyMakeUnsigned<int>{typedef unsigned int type;};
+   template <> struct MyMakeUnsigned<long int>{typedef unsigned long int type;};
++  template <> struct MyMakeUnsigned<long long int>{typedef unsigned long long int type;};
+   template <> struct MyMakeUnsigned<__int128>{typedef unsigned __int128 type;};
```

Comment:
I'm not sure if this makes sense. It seems like support for `__int128` is patchy and on some systems it's just `long long int`, at least that's my guess from errors like
```cpp
src/gfanlib_circuittableint.h:617:57: error: invalid use of incomplete type 'struct gfan::MyMakeUnsigned<long long int>'
  617 |                 typename MyMakeUnsigned<longword>::type positiveResultBoundTimesD=extMul(negabs(s).v,boundA.v);
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
```
It's a mystery to me why having both declarations makes the compiler happy, but I hope it's not introducing an overflow bug.

Reply:
These new `__int128`'s are giving me lots of problem in the Debian package, too -- gfan won't even build on 32-bit systems any more.